css-view-transitions-2/Overview.bs

```diff
@@ -434,32 +434,44 @@ The {{CSSViewTransitionRule}} represents a ''@view-transition'' rule.
 ### Setting up the view-transition in the old {{Document}} ###  {#setup-old-document-vt}
 
 	<div algorithm>
-		To <dfn export>setup cross-document view-transition</dfn> given a {{Document}} |oldDocument|,
-		a {{Document}} |newDocument|, a {{NavigationType}} |navigationType|, a boolean |isBrowserUINavigation|, and |onReady|, which is an algorithm accepting nothing:
+		To check if a <dfn export>should navigation trigger a cross-document view-transition?</dfn> given
+		an [=origin=] |oldOrigin|, an [=origin=] |newOrigin|, a boolean |newDocumentWasCreatedWithCrossOriginRedirects|, a {{NavigationType}} |navigationType|, and a boolean |isBrowserUINavigation|:
 
 		1. If the user agent decides to display an [=implementation-defined=] navigation experience, e.g. a gesture-based transition for a back navigation,
-			the user agent may ignore the author-defined view transition. If that is the case, return.
+			the user agent may ignore the author-defined view transition. If that is the case, return false.
+
+		1. If |navigationType| is {{NavigationType/reload}}, then return false.
 
-		1. If |navigationType| is {{NavigationType/reload}}, then return.
+		1. If ||newDocumentWasCreatedWithCrossOriginRedirects|, then return false.
 
-		1. If |isBrowserUINavigation| is true, and |navigationType| is {{NavigationType/push}} or {{NavigationType/replace}}, then return.
+		1. If |isBrowserUINavigation| is true, and |navigationType| is {{NavigationType/push}} or {{NavigationType/replace}}, then return false.
 
-		1. If |oldDocument|'s [=environment settings object/origin=] is not [=same origin=] as
-			|newDocument|'s [=environment settings object/origin=] then call |onReady| and return.
+		1. If |oldOrigin| is not [=same origin=] as |newOrigin| then return false.
 
 		1. If |newDocument| [=was created via cross-origin redirects=] is true
```

Comment:
There is an early return above for the "created with cross-origin redirects" case so we'll never hit this.

Reply:
Not exactly, but I'll move this bit to the HTML spec PR. A navigation was created with a cross-origin redirects if:
- the document was created with cross-origin redirects
- this is not a BFCache restore